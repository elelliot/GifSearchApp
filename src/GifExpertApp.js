
import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';




const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState( defaultCategories );

    // const handleAdd = () => {

    //     // setCategories( 'HunterXHunter' ); esto cambia de el estado de arreglo por un string osea es incorrecto

    //     //usamos spread operator para mantener las categorias ya que es const y agregamos las nuevas
    //     // setCategories( [...categories, 'HunterXHunter'] );


    //     //también puede tener un callback que tiene como argumento el valor del estado anterior y regresa el nuevo
    //     setCategories( cats => [...cats, 'HunterXHunter'] ); 
    // }
    
    
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr />

            

            <ol>
                {
                    categories.map( category => ( // iteracion, indice (no lo pusimos)
                        // return <li key={ category }> { category } </li>//estos keys suelen ser id´s de bases de datos, osea únicos
                         <GifGrid 
                            key = { category }
                            category = { category } />//no usamos el return por que solo regresa una cosa en el otro también lo podemos quitar
                    ))
                }
            </ol>
        </>
    )
}


export default GifExpertApp

//rafcp+tab Functional Component Template