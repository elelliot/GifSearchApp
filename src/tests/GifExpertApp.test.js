import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import GifExpertApp from "../GifExpertApp";


describe('Pruebas en <GifExpertApp />', () => {
    



    test('Debe mostrar bien el comp', () => {

        const wrapper = shallow( <GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();
        
    });


    test('Debe mostrar una lista de categorias ', () => {
        //no hay manera directa de ponerle un valor por defecto en el useState, por lo que lo hicimos el arreglo de valores en el comp
        
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories = { categories } /> );

        expect( wrapper ).toMatchSnapshot()
        
        //Que el numero de elementos GifGrid sean igual al numero de categorias por default 
        expect( wrapper.find('GifGrid').length ).toBe( categories.length ); 
    })
    
    
})
