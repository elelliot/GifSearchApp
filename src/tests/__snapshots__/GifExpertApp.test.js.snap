// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe mostrar bien el comp 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> Debe mostrar una lista de categorias  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
